### DISCUSSION TITLE
Printing in MAUI

### DISCUSSION THREAD

**Discussion by SunshineSpring666** [CommunityToolkit/Maui#562] [New Feature Discussions]
MAUI is awesome. Some OA and LOB apps need basic functionalities like printing files (documents, images, receipts, etc.). Existing code which works for Xamarin raises puzzling exceptions when migrated to MAUI projects. 
To my knowledge, there are no such libs for MAUI now. Any clues on how to get "printing by printer" functionality in MAUI (on Windows/Android/iOS)?

Thanks to the great Syncfusion team, a [Xamarin Printing demo](https://www.syncfusion.com/kb/8767/how-to-print-pdf-documents-in-xamarin-forms-platform) with [sample project](https://www.syncfusion.com/downloads/support/directtrac/general/ze/PDFPrintingSample1519956005.zip) could be found, it works basically ok in Xamarin, but when migrated to MAUI, the Windows code raises exceptions like "Interface not registered", "Class not registered", etc. Some issues are solved by replacing UI Dispatcher with  Dispatcher Queue, and using PrintManagerInterop; while still no clues on these "xxx not registered" issues. These issues seems to be raised due to   switching to WindowsAppSDK WinUI3 in MAUI?

Thanks.

**Comment by bijington** [comment]
What was possible on iOS and Android should still work, there will be differences in how to implement the bits as you won't need the `DependencyService` in the .NET MAUI implementation. As for the windows exceptions I really don't know. You will have to dig into what classes it claims aren't implemented and what the WinUI alternatives are.

In summary printing will be possible but it doesn't come out of the box from the .NET MAUI team. You could ask this question on the .NET MAUI repo to see if they have any thoughts on it.

**Comment by SunshineSpring666** [comment]
Hi @bijington,
Actually, the Android implementation crashes with exception, too. 
The demo works ok on Xamarin, but doesn't on MAUI. I don't know what deep down inside got changed from this Xamarin-to-MAUI migration. But as app developers, we expect such basic reusable functionalities could be supported as a NuGet lib, so that we can focus on implementing app logic code.
Thanks for your reply.
Best Regards

**Comment by CliffAgius** [comment]
Closed as answered
